### PR TITLE
Sample module

### PR DESCRIPTION
**Pull request by cymon1997**
Implement sample end to end from repository to handler

### CODE REVIEW

File: internal/errors/entity.go [hunk 1]
```diff
@@ -8,4 +8,10 @@ var (
 		Code:    "ERROR_UNKNOWN",
 		Message: "Unexpected error",
 	}
+
+	ErrDatabaseOp = &Err{
+		Status:  http.StatusInternalServerError,
+		Code:    "ERROR_DATABASE_OPERATION",
```
Comment (cymonevolution) [inline internal/errors/entity.go:14]: 👍

File: core/usecase/sample.create.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+	"github.com/google/uuid"
+	"gopkg.in/guregu/null.v4"
+)
+
+func (uc *usecaseImpl) CreateSample(ctx context.Context, input spec.CreateSampleRequest) (*spec.CreateSampleResponse, error) {
+	output, err := uc.dbRepo.InsertSample(ctx, model.Sample{
+		UUID:       uuid.New().String(),
+		Data:       null.StringFrom(input.Sample.Data),
+		CreateTime: null.TimeFrom(time.Now().UTC()),
+		// TODO: Add request source to context data
```
Comment (cymonevolution) [inline core/usecase/sample.create.go:19]: better to put this as an issue to be fixed later on

File: core/outbound/db/sample.insert.go [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+	"context"
+	"log"
+
+	"github.com/cymon1997/go-template/core/outbound/db/model"
+	"github.com/cymon1997/go-template/internal/errors"
+)
+
+func (r *repositoryImpl) InsertSample(ctx context.Context, data model.Sample) (*model.Sample, error) {
+	if err := r.db.Instance().WithContext(ctx).Create(&data).Error; err != nil {
+		log.Print("error insert: ", err)
```
Comment (cymonevolution) [inline core/outbound/db/sample.insert.go:13]: I think we can make constant for error logs so we have unified format

Comment (cymon1997) [reply core/outbound/db/sample.insert.go:13]: ah sure, we can put that in our improvements, thanks

File: core/usecase/sample.create.go (hunk 2 above)
Comment (cymon1997) [reply core/usecase/sample.create.go:19]: okay sure

**Review by progrevo — APPROVED** [review]
LGTM!

**Pull request merged**